### PR TITLE
Corrector

### PR DESCRIPTION
**Pull request by Ritu-Devnani**
Please review the files uploaded for training the corrector.

### CODE REVIEW

File: fairseq_utility.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
+import shlex
+import shutil
```
Comment (akanshak984) [inline fairseq_utility.py:2]: Where have we used `shutil` in this file? If not let's remove this line

File: fairseq_utility.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        if restore_file is not None:
+           cmd += f"--restore-file {restore_file} "
+        if reset:
+           cmd += "--reset-optimizer \
+                   --reset-lr-scheduler \
+                   --reset-dataloader \
+                   --reset-meters "
+        cmd += additional_cmds
+        if logfile is not None:
+            import socket, os
```
Comment (akanshak984) [inline fairseq_utility.py]: We have imported os in the beginning of the file. Do we need to import it here again?

File: fairseq_utility.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
+import shlex
+import shutil
+import subprocess
+import json, os
```
Comment (akanshak984) [inline fairseq_utility.py]: Have we used `json` anywhere?

File: fairseq_utility.py [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+from pathlib import Path
+from collections import defaultdict, OrderedDict
```
Comment (akanshak984) [inline fairseq_utility.py:6]: Please verify all the imports and if it's not used let's remove it.

File: fairseq_utility.py (hunk 1 above)
Comment (Ritu-Devnani) [reply fairseq_utility.py:2]: shutil is used as a pre requisite as many files are generated after firing a commands related to fairseq. This library makes files copy, create, etc operations easier and faster.

Comment (Ritu-Devnani) [reply fairseq_utility.py:2]: shutil is used as a pre requisite as many files are generated after firing a commands related to fairseq. This library makes files copy, create, etc operations easier and faster.

File: fairseq_utility.py (hunk 2 above)
Comment (Ritu-Devnani) [reply fairseq_utility.py]: Done the required changes.

File: fairseq_utility.py (hunk 3 above)
Comment (Ritu-Devnani) [reply fairseq_utility.py]: json is required afterwards for the functions defined for evaluation for easy understanding. I will remove that for now.

File: fairseq_utility.py (hunk 4 above)
Comment (Ritu-Devnani) [reply fairseq_utility.py:6]: subprocess : used for visualising the child processes running while a call to the fairseq_train() function. 
json : Initially removed
os : used for accessing the working directories and retrieve the id of the processes that are running ( this helps us to see the total number of tokens visited, etc. )
Path : used to indicate that a google drive path to which we need to store the output files generated.
collections : the token vocabulary taken as an input through parameter in the above function defined ( "srcdict" ) is stored and used in thee form of a dictionary.

**Review by akanshak984 — APPROVED** [review]

**Comment by akanshak984** [comment]
Avoid commit messages like 
- *required changes* - It gives no idea of what required changes. specify exact changes if it is short or else summarize the changes.
- *thanks* or anything which doesn't give idea of what changes were done in the commit

example:
```
Update fairseq_utility.py
I have done the required changes. Thanks for pointing it out.
```

It gives no idea of what required changes are done.

instead you could write:

```
Update fairseq_utility.py

Remove duplicate import of os

```

**Comment by akanshak984** [comment]
Update the commit message and then we can merge this PR

**Pull request merged**